docker/unit-tests/Dockerfile.base.in

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Intel Corp.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// install GO version set by argument or fallback to default ver.
+ARG UT_GO_VERSION=1.14.3
+RUN wget "https://dl.google.com/go/go${UT_GO_VERSION}.linux-amd64.tar.gz" && \\
+    tar -C "/usr/local" -xzf "go${UT_GO_VERSION}.linux-amd64.tar.gz" && \\
+    rm "go${UT_GO_VERSION}.linux-amd64.tar.gz"
+ENV PATH=$PATH:/usr/local/go/bin
+ENV GOPATH="/root/go"
+ENV GOROOT="/usr/local/go"
+RUN mkdir $GOPATH
+
+// install unit test specific go modules
+RUN go get -u "github.com/msoap/go-carpet"
+RUN ln -s $(go env GOPATH)/bin/go-carpet /usr/local/bin/go-carpet
+
+// copy userspace CNI plugin code and tests from host to container
+RUN mkdir -p $GOPATH/src/github.com/intel/userspace-cni-network-plugin
+COPY ./ $GOPATH/src/github.com/intel/userspace-cni-network-plugin/
```

Comment:
In my testing, looks like to run the tests, `make` needs to be run on the host so when this copy is performed the `./cnivpp/bin_api/*` directories and files are in place. This either needs to documented in the README.md, or my preference would be to run `make` here.

Here is the error I was getting when `make` on the host was not run:
```
$ make test-centos7
Run unit tests at centos7
go: downloading github.com/containernetworking/cni v0.8.0
go: downloading k8s.io/api v0.17.0
:
=== RUN   TestWritePodAnnotation/fail_to_get_pod
2020-10-09T17:19:41Z [error] status update failed for pod namespace-d104e7d9/pod-d104e7d9: pods "pod-d104e7d9" not found
--- PASS: TestWritePodAnnotation (0.00s)
    --- PASS: TestWritePodAnnotation/write_annotations (0.00s)
    --- PASS: TestWritePodAnnotation/fail_with_pod_set_to_nil (0.00s)
    --- PASS: TestWritePodAnnotation/fail_with_kubeClient_set_to_nil (0.00s)
    --- PASS: TestWritePodAnnotation/fail_to_get_pod (0.00s)
PASS
ok  	github.com/intel/userspace-cni-network-plugin/pkg/k8sclient	0.015s
/root/go/src/github.com/intel/userspace-cni-network-plugin
go: downloading github.com/containernetworking/plugins v0.8.0
go: downloading github.com/vishvananda/netlink v1.1.0
go: downloading git.fd.io/govpp.git v0.3.5
go: downloading github.com/sirupsen/logrus v1.7.0
go: finding module for package github.com/intel/userspace-cni-network-plugin/cnivpp/bin_api/l2
go: downloading github.com/vishvananda/netns v0.0.0-20191106174202-0a2b9b5464df
go: finding module for package github.com/intel/userspace-cni-network-plugin/cnivpp/bin_api/memif
go: downloading github.com/fsnotify/fsnotify v1.4.7
go: finding module for package github.com/intel/userspace-cni-network-plugin/cnivpp/bin_api/interfaces
go: downloading github.com/lunixbochs/struc v0.0.0-20200707160740-784aaebc1d40
go: downloading github.com/coreos/go-iptables v0.4.5
go: downloading github.com/safchain/ethtool v0.0.0-20200804214954-8f958a28363a
../cnivpp/api/interface/interface.go:28:2: no matching versions for query "latest"
../cnivpp/api/bridge/bridge.go:26:2: no matching versions for query "latest"
../cnivpp/api/memif/memif.go:28:2: no matching versions for query "latest"
make: *** [test-centos7] Error 1
```

Reply:
hey @Billy99 I managed to reproduce that one - I added VPP deps installation and VPP binapi code generation as dependencies of the `make test` target, could you please confirm whether that works for you?

Reply:
I pushed it too quickly it seems, test-all fails, I'll figure out something smarter and update the PR

Reply:
I was just comping an email to let you know it was still failing for me.

Reply:
can you please take another look? I moved install-dep, install and default target to Dockerfile so that correct distro-specific VPP API packages and generate API code are always present in the test container images - this should work better than copying them from the host

Reply:
centos7 and ubuntu20.04 worked properly. Thanks for the fix.